src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientMethodsAnalyzer.cs

```diff
@@ -185,19 +187,44 @@ private static void CheckProtocolMethodParameters(ISymbolAnalysisContext context
             }
 
             var requestContent = method.Parameters.FirstOrDefault(p => p.Type.Name == "RequestContent");
-            if (requestContent == null)
+            if (requestContent == null && method.Parameters.Last().IsOptional)
             {
-                if (method.Parameters.LastOrDefault().IsOptional)
+                INamedTypeSymbol type = (INamedTypeSymbol)context.Symbol;
+                var methodList = type.GetMembers(method.Name).OfType<IMethodSymbol>().Where(member => !SymbolEqualityComparer.Default.Equals(member, method));
+                var convenienceMethod = FindMethod(methodList, method.TypeParameters, method.Parameters.RemoveAt(method.Parameters.Length - 1), symbol => IsCancellationToken(symbol), ParameterEquivalenceComparerOptionalIgnore.Default);
+                if (convenienceMethod != null)
                 {
                     context.ReportDiagnostic(Diagnostic.Create(Descriptors.AZC0018, method.Locations.FirstOrDefault()), method);
                 }
             }
             // Optional RequestContent or required RequestContent is checked in AnalyzeCodeBlock.
         }
 
-        // A protocol method should not have model as type. Accepted return type: Response, Task<Response>, Pageable<BinaryData>, AsyncPageable<BinaryData>, Operation<BinaryData>, Task<Operation<BinaryData>>, Operation, Task<Operation>
+        // A protocol method should not have model as type. Accepted return type: Response, Task<Response>, Pageable<BinaryData>, AsyncPageable<BinaryData>, Operation<BinaryData>, Task<Operation<BinaryData>>, Operation, Task<Operation>, Operation<Pageable<BinaryData>>, Task<Operation<AsyncPageable<BinaryData>>>
         private static void CheckProtocolMethodReturnType(ISymbolAnalysisContext context, IMethodSymbol method)
         {
+            bool IsValidPageable(ITypeSymbol typeSymbol)
+            {
+                if (typeSymbol.Name != "Pageable" && typeSymbol.Name != "AsyncPageable")
```

Comment:
This feels brittle.  What happens if it's a derrived type or we rename?   At the very least, let's please abstract all these magic strings into a constants file.   This set, for example, is used at least three times.

Reply:
Not blocking for this PR, but let's please open an issue or follow-up to refactor this.

Reply:
Why a protocol method would return a derived type or we would return? They are all both breaking change.

Reply:
Already fixed in this PR.